File: backend/src/main/java/ru/chsergeig/shoppy/service/admin/AdminAccountService.java
package ru.chsergeig.shoppy.service.admin;

import ru.chsergeig.shoppy.dto.admin.AccountDto;

import java.util.List;

public interface AdminAccountService {

    List<AccountDto> getAllAccounts();

    AccountDto getAccountByLogin(String login);

    AccountDto addAccount(String name);

    AccountDto addAccount(AccountDto dto);

    AccountDto updateAccount(AccountDto dto);

    Integer deleteAccount(String name);

}

File: backend/src/main/kotlin/ru/chsergeig/shoppy/service/admin/AdminAccountService.kt
package ru.chsergeig.shoppy.service.admin

import ru.chsergeig.shoppy.model.AdminAccountDto

interface AdminAccountService {

    fun getAllAccounts(): List<AdminAccountDto?>

    fun getAccountByLogin(
        login: String?
    ): AdminAccountDto

    fun addAccount(
        name: String?
    ): AdminAccountDto

    fun addAccount(
        dto: AdminAccountDto?
    ): AdminAccountDto

    fun updateAccount(
        dto: AdminAccountDto?
    ): AdminAccountDto

    fun deleteAccount(
        name: String?
    ): Int
}
